### ISSUE TITLE
implement 'user display' component for conveying name, username, role, and other information consistently

### ISSUE THREAD

**Issue by indirectlylit** [learningequality/kolibri#5024]
### Observed behavior

In a coach report, these are two different users:

> ![image](https://user-images.githubusercontent.com/2367265/52674944-fc77c780-2ed9-11e9-9f8c-47a95f4f0cb7.png)

In facility users we show:

> ![image](https://user-images.githubusercontent.com/2367265/52674997-28934880-2eda-11e9-9184-1a129e229676.png)

In class management we show:

> ![image](https://user-images.githubusercontent.com/2367265/52675034-43fe5380-2eda-11e9-81aa-feb39c2f8921.png)

related clearinghouse issues:

* https://github.com/learningequality/clearinghouse/issues/185
* https://github.com/learningequality/clearinghouse/issues/299

### Expected behavior

Recommend we come up with a consistent, single-column representation of a user that can be used in all tables. This block would contain:

* full name
* username
* user type

It would be responsive to the available space and to the lengths of the values. It could optionally leverage tooltips or advanced ellipsis behaviors to deal with small screens and long names.

As one possible example, we might try something like:

> ![image](https://user-images.githubusercontent.com/2367265/52675827-e1f31d80-2edc-11e9-9e0f-ea86a05102e1.png)

### User-facing consequences

Hard to scan and differentiate users

### Context

0.12.0 beta 4

**Comment by khangmach** [comment]
I like this a lot and am on board with placing username under fullname as opposed to its own column. Helps make a clear differentiation between name and the more abstract and fluid username. 

However, in your implementation above I think it's much more difficult to:
- scan for role label (super admin/admin/fac. coach/coach) which should be of high value when you're looking for people of specific roles in a long list (for example, enrolling coaches as learners in various contexts). 
- tell the difference between username and role label. 
- makes it little more difficult to copy/paste username text if you want to reference it for something.

I would advocate for placing the role label in its own column, where username was previously

**Comment by indirectlylit** [comment]
yeah makes sense. another dimension to possibly play with is the icon... we do this in facility class management:

> ![image](https://user-images.githubusercontent.com/2367265/53130760-fa3beb80-351f-11e9-83f3-0f0bd9f7914b.png)

and we do this in device permission management:

> ![image](https://user-images.githubusercontent.com/2367265/53130789-0cb62500-3520-11e9-8cfe-c1a90a29e46b.png)

so maybe there's some kind of consistent role-based iconography we could establish that would negate the need for an additional column

**Comment by khangmach** [comment]
example

![table](https://user-images.githubusercontent.com/25045166/53131161-ddec7e80-3520-11e9-9938-501840549685.png)

**Comment by khangmach** [comment]
>maybe there's some kind of consistent role-based iconography we could establish that would negate the need for an additional column

Yeah, it's a bit of a tricky problem because we strictly use google material icons, but I don't see much issue in swapping a user icon for a key/coach icon. We still convey the concept of "what is a user, and what info is associated with a user" very clearly (albeit repetitively) through tables throughout the app anyways.

**Comment by indirectlylit** [comment]
just an idea:

> ![image](https://user-images.githubusercontent.com/2367265/53135513-54dd4380-3530-11e9-8d22-1ca8e33b712c.png)

**Comment by khangmach** [comment]
that works to me 👀 thoughts @jtamiace ?

**Comment by jtamiace** [comment]
Works for me too. We'll just need to find an icon for class coach with this approach (could be an unfilled version of the one above?), and facility admin (and class admin down the line if that's ever a thing)

**Comment by indirectlylit** [comment]
We'll still need to put a little more thought into a couple things about context... e.g.:

* is it useful/necessary to show device permissions in all contexts?
* do we show icons for the user's role in a class, their user type, or both
* ... others?

**Comment by khangmach** [comment]
![slice](https://user-images.githubusercontent.com/25045166/53845759-2ebb9880-3f5f-11e9-8f85-60e75db713f7.png)


Some solutions based on our discussions over slack.
Suggested high level improvements:
- Give key icons their own column and color coordinate them based on permission level
- Separate the role label from the username column and give the role label it's own column
- Introduce admin icon to be consistent with coach icon

**Comment by indirectlylit** [comment]
this looks great - nice job!

minor clarification: I assume in the table on the right, all "Super admin" users should have gold keys?

**Comment by khangmach** [comment]
@indirectlylit whoops yes. Adjusted the original image
